packages/core/test/feedback/ScreenshotButton.test.tsx

```diff
@@ -0,0 +1,216 @@
+      enableTakeScreenshot: true,
+    });
+    getClient()?.addIntegration(integration);
+
+    showFeedbackButton();
+
+    fireEvent.press(getByText('Report a Bug'));
+
+    const takeScreenshotButton = getByText('Take a screenshot');
+    expect(takeScreenshotButton).toBeTruthy();
+  });
+
+
+  it('the capture screenshot button is shown when tapping the Take a screenshot button in the feedback widget', async () => {
+    const { getByText } = render(
+      <FeedbackWidgetProvider>
+        <Text>App Components</Text>
+      </FeedbackWidgetProvider>
+    );
+
+    const integration = feedbackIntegration({
+      enableTakeScreenshot: true,
+    });
+    getClient()?.addIntegration(integration);
+
+    showFeedbackButton();
+
+    fireEvent.press(getByText('Report a Bug'));
+    fireEvent.press(getByText('Take a screenshot'));
+
+    const captureButton = getByText('Take Screenshot');
+    expect(captureButton).toBeTruthy();
+  });
+
+  it('a screenshot is captured when tapping the Take Screenshot button', async () => {
+    mockCaptureScreenshot.mockResolvedValue([mockScreenshot]);
+    mockEncodeToBase64.mockResolvedValue(mockBase64Image);
+
+    const { getByText } = render(
+      <FeedbackWidgetProvider>
+        <Text>App Components</Text>
+      </FeedbackWidgetProvider>
+    );
+
+    const integration = feedbackIntegration({
+      enableTakeScreenshot: true,
+    });
+    getClient()?.addIntegration(integration);
+
+    showFeedbackButton();
+
+    fireEvent.press(getByText('Report a Bug'));
+    fireEvent.press(getByText('Take a screenshot'));
+    fireEvent.press(getByText('Take Screenshot'));
+
+    await waitFor(() => {
+      expect(mockCaptureScreenshot).toHaveBeenCalled();
+    });
+
+    await waitFor(() => {
+      expect(mockEncodeToBase64).toHaveBeenCalled();
+    });
+  });
+
+  it('the feedback widget ui is updated when a screenshot is captured', async () => {
+    mockCaptureScreenshot.mockResolvedValue([mockScreenshot]);
+    mockEncodeToBase64.mockResolvedValue(mockBase64Image);
+
+    const { getByText, queryByText } = render(
+      <FeedbackWidgetProvider>
+        <Text>App Components</Text>
+      </FeedbackWidgetProvider>
+    );
+
+    const integration = feedbackIntegration({
+      enableTakeScreenshot: true,
+    });
+    getClient()?.addIntegration(integration);
+
+    showFeedbackButton();
+
+    fireEvent.press(getByText('Report a Bug'));
+    fireEvent.press(getByText('Take a screenshot'));
+    fireEvent.press(getByText('Take Screenshot'));
+
+    await waitFor(() => {
+      expect(mockCaptureScreenshot).toHaveBeenCalled();
+    });
+
+    await waitFor(() => {
+      expect(mockEncodeToBase64).toHaveBeenCalled();
+    });
+
+    await waitFor(() => {
+      const captureButton = queryByText('Take Screenshot');
+      expect(captureButton).toBeNull();
+    });
+
+    await waitFor(() => {
+      const takeScreenshotButtonAfterCapture = queryByText('Take a screenshot');
```

Comment:
`Take a screenshot` not rendered could also be because the widget has not rendered after taking the screenshot. Would it make sense to check for the `Remove screenshot` button?

Reply:
Good point Krystof 👍 Added with 8876d89e50d814270cab338d3feb96e1355b4082
For some reason this was flaky 885399c5300003f787327dd9c488b399dccf9f8d in my original tests but it seems to work now :) 🤞

Reply:
Thanks for the context; I didn't know that.